src/utils/array/isUniqueArrayItem/isUniqueArrayItem.ts

```diff
@@ -3,4 +3,10 @@ export const isUniqueArrayItem = (array: string[]): boolean => {
   return uniqueArrayItems.size === array.length;
 };
 
-export const isUniqueArrayItemWithNewEntry = (array: string[], name: string): boolean => isUniqueArrayItem([...array, name]);
+export const isUniqueArrayItemWithNewEntry = (array: string[], entry: string, excludeEntry?: string | false | undefined): boolean => {
```

Comment:
Oulala les `v` je veux pas voir ça 🙈😀

Plutôt faire du déclaratif 

```suggestion
export const isUniqueArrayItemWithNewEntry = (array: string[], entry: string, excludeEntry?: string | false | undefined): boolean => {
  const arrayNames = [...array, entry];

  if (excludeEntry) {
    isUniqueArrayItem(arrayNames.filter((item) => item !== excludeEntry));
  }

  return isUniqueArrayItem(arrayNames);
};
```